lib/Parse/ParseExpr.cpp

```diff
@@ -421,8 +421,9 @@ ParserResult<Expr> Parser::parseExprSequenceElement(Diag<> message,
 
   if (Tok.isContextualKeyword("unsafe") &&
       !(peekToken().isAtStartOfLine() ||
-        peekToken().isAny(tok::r_paren, tok::r_brace, tok::r_square,
-                          tok::equal, tok::colon, tok::comma, tok::eof) ||
+        peekToken().isAny(tok::r_paren, tok::r_brace, tok::r_square, tok::equal,
+                          tok::colon, tok::comma, tok::eof,
+                          tok::question_infix) ||
```

Comment:
Rather than checking for `tok::question_infix` I think it would be better to check for `peekToken().isBinaryOperatorLike()` since we shouldn't parse e.g `unsafe + 1` as an `unsafe` expression either. It would also be great if you could add a check for `isPostfixOperatorLike` to handle e.g `unsafe!`.

Ideally we ought to be matching [the syntax-syntax logic](https://github.com/swiftlang/swift-syntax/blob/c2eabd2ee212e5c53058f3bd5228aabebb362792/Sources/SwiftParser/Statements.swift#L918) as closely as we can here.

Reply:
@hamishknight 
Thank you very much! I've fixed it. [df15f94](https://github.com/swiftlang/swift/pull/81429/commits/df15f94292eef867c60d730448d3789607fffebd)

> Ideally we ought to be matching [the syntax-syntax logic](https://github.com/swiftlang/swift-syntax/blob/c2eabd2ee212e5c53058f3bd5228aabebb362792/Sources/SwiftParser/Statements.swift#L913) as closely as we can here

That makes sense. I’ll keep it in mind as a possible improvement for later.

Reply:
Thanks! One more comment but otherwise this LGTM